cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/SlangCompilerImpl.java

```diff
@@ -102,10 +102,10 @@ public ExecutableModellingResult preCompileSource(SlangSource source) {
         Validate.notNull(source, "You must supply a source to compile");
 
         //first thing we parse the yaml file into java maps
-        ParsedSlang parsedSlang = yamlParser.parse(source);
```

Comment:
it is a little not intuitive to have a method called `parseAndValidate`, a method should do only one thing in general. Can it be that parse and validate can be separated without introducing performance problems? If that is not possible maybe a better name is `validate` as it is already called on a yamlParser